src/System.Collections.Specialized/src/System/Collections/Specialized/NameValueCollection.cs

```diff
@@ -43,11 +44,11 @@ public NameValueCollection() : base()
         public NameValueCollection(NameValueCollection col)
             : base(col != null ? col.Comparer : null)
         {
-            Add(col);
+            Add(col!);
```

Comment:
Add method throws on null, was not able to use Debug.Assert(col != null); from test failure

Reply:
Yes here of course you can't use `Debug.Assert(col !=null)` because it is a public API where users can pass down a `null` value and we don't control that, that is why we throw in the `Add` method. So here, your only option is what you did, which is bang it away, as `Add` clearly can't accept `null`. In this case, we bang it away and let it flow because `Add` is doing the validation. 

The reason why you need to bang it away, is because the check we use when doing the `base` call, is changing the nullability state for the compiler because we compared it against `null`, so the compiler says, oh, even though it is not marked as a nullable value, it is being compared against `null`, then it should be nullable afterwards.

Reply:
Good explanation, thanks!